guides/keras_cv/semantic_segmentation_deeplab_v3_plus.py

```diff
@@ -0,0 +1,298 @@
+    num_classes=21,
+    y_true=batch["segmentation_masks"],
+    scale=3,
+    rows=2,
+    cols=2,
+)
+
+"""
+## Model Configuration
+
+Please feel free to modify the configurations for model training and note how the
+training results changes. This is an great exercise to get a better understanding of the
+training pipeline.
+
+The learning rate schedule is used by the optimizer to calculate the learning rate for
+each epoch. The optimizer then uses the learning rate to update the weights of the model.
+In this case, the learning rate schedule uses a cosine decay function. A cosine decay
+function starts high and then decreases over time, eventually reaching zero. The initial
+learning rate is 0.007 and the decay steps are 2124. This means that the learning rate
+will start at 0.007 and then decrease to zero over 2124 steps.
+"""
+
+BATCH_SIZE = 4
+GLOBAL_BATCH_SIZE = BATCH_SIZE * 1
+INITIAL_LR = 0.007 * GLOBAL_BATCH_SIZE / 16
+EPOCHS = 100
+NUM_CLASSES = 21
+learning_rate = keras.optimizers.schedules.CosineDecay(
+    INITIAL_LR,
+    decay_steps=EPOCHS * 2124,
+)
+
+"""
+Let us create an instance of the `DeepLabV3Plus` model for semantic segmentation. The
+model architecture is initialized with `resnet50_v2_imagenet` preset and 21
+classes. `resnet50_v2_imagenet` pre-trained weights will be used as the backbone feature
+extractor for the DeepLabV3Plus model. `num_classes` parameter specifies the number of
+classes that the model will be trained to segment.
+"""
+
+model = keras_cv.models.DeepLabV3Plus.from_preset(
+    "resnet50_v2_imagenet", num_classes=NUM_CLASSES
+)
+
+
+"""
+## Compile the model
+
+The model.compile() function sets up the training process for the model. It defines the
+- optimization algorithm - Stochastic Gradient Descent (SGD)
+- the loss function - categorical cross-entropy
+- the evaluation metrics - Mean IoU and categorical accuracy
+
+that will be used during training and validation.
+"""
+
+model.compile(
+    optimizer=keras.optimizers.SGD(
+        learning_rate=learning_rate, weight_decay=0.0001, momentum=0.9, clipnorm=10.0
+    ),
+    loss=keras.losses.CategoricalCrossentropy(from_logits=False),
+    metrics=[
+        keras.metrics.MeanIoU(
+            num_classes=NUM_CLASSES, sparse_y_true=False, sparse_y_pred=False
+        ),
+        keras.metrics.CategoricalAccuracy(),
+    ],
+)
+
+model.summary()
+
+"""
+The utility function `dict_to_tuple` effectively transforms the dictionaries of training
+and validation datasets into tuples of images and one-hot encoded segmentation masks,
+which is used during training and evaluation of the `DeepLabV3Plus` model.
+"""
+
+
+def dict_to_tuple(x):
+    return x["images"], tf.one_hot(
+        tf.cast(tf.squeeze(x["segmentation_masks"], axis=-1), tf.int32), 21
+    )
+
+
+train_ds = train_ds.map(dict_to_tuple)
+eval_ds = eval_ds.map(dict_to_tuple)
+
+model.fit(train_ds, validation_data=eval_ds, epochs=EPOCHS)
+
+"""
+## Prediction with trained model
```

Comment:
How long does it take to train the full model? If it's long we may want to make `model.fit` above just do 1 epoch.

Either way, it'd be great to have a pre-trained preset to show 1-line inference without training (maybe even with the same dog pic from the OD guide).

You could add a preset from this training script in a PR to KerasCV, or I can help out with that if you'd like! It's a nice way to start the guide (the same way we did the OD guide) if we show inference with a pre-trained model.

Reply:
I agree! Showing how to load a model and try inference with a single image and then showing how to fine-tune the model would be a great flow for the guide. I'll train this and add a preset.

Reply:
done!